beir/retrieval/search/dense/exact_search.py

```diff
@@ -70,21 +74,21 @@ def search(self,
             cos_scores[torch.isnan(cos_scores)] = -1
 
             # Get top-k values
-            cos_scores_top_k_values, cos_scores_top_k_idx = torch.topk(cos_scores, min(top_k+1, len(cos_scores[1])), dim=1, largest=True, sorted=return_sorted)
+            cos_scores_top_k_values, cos_scores_top_k_idx = torch.topk(cos_scores, min(top_k, len(cos_scores[1])), dim=1, largest=True, sorted=return_sorted)
```

Comment:
You write that `Which means some queries would have top_k retrieved documents, while others have top_k-1 retrieved documents.`, but didn't this `+1` ensure that that does not happen cuz we retrieve `top_k+1` but then only allow `top_k` lateron?

Reply:
IIUC, the problem comes from this line
https://github.com/beir-cellar/beir/blob/505d80dd85c8d2660a73a814cd2c56ba487e5661/beir/retrieval/search/dense/exact_search.py#L86
So we only keep the top_k (which sometimes include the query inside the retrieved docs)

Reply:
I see, I thought the `if corpus_id != query_id:` would ensure that the query would never be added to `result_heaps[query_id]` 🧐

Reply:
Hmm, then why do we get different results? 🧐

Reply:
It's easy to check, we just have to assert that number of results of each query is `top_k`. Can you check that please @Muennighoff ?